### ISSUE TITLE
twisted/internet/test/test_process.py tests which might fail while waiting for an event should use runReactor instead of reactor.run

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3426]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3426|
|Type|defect|
|Created|2008-09-10 14:56:53Z|
|Branch|https://github.com/twisted/twisted/tree/runreactor-3426|

`ReactorBuilder.runReactor` implements the trial timeout.  `reactor.run` will go forever if it isn't stopped.  If a test is waiting for an event before it stops the reactor, if the implementation is buggy and the event never arrives, the test will spin forever.

This is probably related to [#3398](https://github.com/twisted/twisted/issues/3398).

<details><summary>Searchable metadata</summary>

```
trac-id__3426 3426
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_runreactor_3426 branches/runreactor-3426
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1221058613000000 1221058613000000
changetime__1230837885000000 1230837885000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [24774]) Branching to 'runreactor-3426'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [24775]) switch process-running tests to runReactor instead of direct reactor.run calls

refs [#3426](https://github.com/twisted/twisted/issues/3426)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set status to `closed`|
|-|-|

This was fixed in r25366 as part of the [#3521](https://github.com/twisted/twisted/issues/3521) branch.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
